src/stdlib/SDL_string.c

```diff
@@ -837,6 +837,43 @@ char *SDL_strstr(const char *haystack, const char *needle)
 #endif /* HAVE_STRSTR */
 }
 
+char *SDL_strnstr(const char *haystack, const char *needle, size_t size)
+{
+#ifdef HAVE_STRNSTR
+    return SDL_const_cast(char *, strnstr(haystack, needle, size));
+#else
+    size_t needlesize;
+    const char *end, *giveup;
+
+    needlesize = strlen(needle);
+    /* We can give up if needle is empty, or haystack can never contain it */
+    if (needlesize == 0 || needlesize > size)
+       return NULL;
+    /* The pointer value that marks the end of haystack */
+    end = haystack + size;
+    /* The maximum value of i, because beyond this haystack cannot contain needle */
+    giveup = end - needlesize + 1;
```

Comment:
Sanitizers _may_ choke on this in the for construct, i.e. at `i != giveup` because you may go out-of-bounds?